tensornetwork/backend_contextmanager.py

```diff
@@ -0,0 +1,17 @@
+from typing import Text, Union
+from tensornetwork.backends.base_backend import BaseBackend
+
+class DefaultBackend():
+
+    with_statement = False
+    backend = None
+
+    def __init__(self, backend: Union[Text, BaseBackend]) -> None:
+        DefaultBackend.backend = backend
+
+    def __enter__(self):
+        DefaultBackend.with_statement = True
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        DefaultBackend.with_statement = False
+        DefaultBackend.backend = None
```

Comment:
Nice! I like the object oriented approach.

However, I think we should be using a stack based approach for finding the current backend. (For an example, see what we did with the `NodeCollection` [code](https://github.com/google/TensorNetwork/blob/83ff4af11dcd03d6c0db70e139c7137154dc015d/tensornetwork/network_components.py#L1978).